### ISSUE TITLE
Issue with the nested styles (third level)

### ISSUE THREAD

**Issue by naxxateux** [postcss/postcss-nested#125]
Hi!

Just updated postcss-nested to v. 5.0.5 (from 5.0.3) and got this issue.

This is my .sss file:
```
...
.text
  ...
  &:hover
    border-color: red
      &:before
        color: red
  ...
```

And I got this as a result:
```
.text:hover {
    border-color: red;
    &: :before {;
    color: red;
    }: ;
}
```

What seems to be the problem here? And how can I fix it properly? Thnx in advance.

**Comment by ai** [comment]
Seems like a strange bug in the new PostCSS 8 event system.

I will have a relocation this week, so if you want to fix it quicker, try to investigate what is happening here https://github.com/postcss/postcss-nested/blob/main/index.js#L143-L155

**Comment by naxxateux** [comment]
@ai thnx for the quickest response ever :-) I will try to investigate it next week.

**Comment by TheSeally** [comment]
@naxxateux, I can't reproduce this issue. I tried it on windows and ubuntu with different postcss parsers and bundlers, but css was always processed correctly.
Can you add additional info about your environment or maybe try to reproduce issue again?

**Comment by ai** [comment]
There is also similar issue here https://github.com/postcss/postcss-mixins/issues/132

**Comment by ai** [comment]
And seems like here we have similar issue https://github.com/postcss/postcss-nested/issues/129

**Comment by TheSeally** [comment]
I reproduced issue.
It is very interesting that with `postcss-nested@5.0.5` and `postcss@8.3.0`  there is no problem, but with `postcss@8.3.1` issue happened.
I'll debug it and try to make PR for fix

**Comment by ai** [comment]
@TheSeally very, very strange. `postcss@8.3.1` changes only `no plugins or parser` warning detection.

Maybe `node_modules` structure somehow creates a bug.

**Comment by exdeniz** [comment]
I added tests and check on these versions of postcss:
8.3.2 
8.3.1
8.3.0
8.2.15 
8.2.0
8.1.3
and 
and all version postcss-selector-parser@6.0.x,
and node 10.21, 12.18.2, 14.10.1, 16.3.0
All test passed.

**Comment by ai** [comment]
@exdeniz it means that issue is with some another component. Maybe PostCSS?

**Comment by exdeniz** [comment]
@ai I had a suspicion that maybe postcss-nested was preceded by postcss-present-env with the postcss-nesting plugin. I investigated for two issues and #129  is suspiciously similar to the result obtained by postcss-nestign.
```
/* postcss-nesting */

.selector {
  :global {
    h2 {
      color: pink;
    }
  }
}

/* postcss-nested */

.selector :global h2 {
  color: pink;
}
```

**Comment by ai** [comment]
Hm. We can add a warning to `postcss-nested` if we will find `postcss-nesting` in `result.processor.plugins`

**Comment by ai** [comment]
@naxxateux did we have `postcss-preset-env` in Amplifr?

**Comment by exdeniz** [comment]
@naxxateux And you can get a config PostCSS?

**Comment by TheSeally** [comment]
I reproduced it in https://github.com/ai/sitnik.ru project, but it reproduced only when I use `yarn`

Steps:
1. clone and install deps via `yarn` (now postcss version is 8.3.0)
2. add nested styles
3. build project (now css processed corectly)
4. update postcss to 8.3.1 via `yarn`
5. try to build again (now styles compile with problem. also bug with media queries syntax with `< >` operators)

When I install deps via npm there is no problem :)

**Comment by ai** [comment]
😅 Very strange bug.

But a few people told me that they have a bug after PostCSS update. Can you find a source of problem in a broken environment?

**Comment by TheSeally** [comment]
yeah I will try to do it

**Comment by TheSeally** [comment]
I debuged a little bit and found that problem can connect with different postcss version in deps.

When I install `postcss@8.3.0` with `sugarss` via `yarn` I have only one `postcss` in deps folder, but when I install `postcss@8.3.1` (via `yarn`) I have two different `postcss` in deps (one standalone package and one in `sugarss` folder).
So because `postcss` use Symbol `isClean` for checking root than when we have 2 different postcss packages we have two different symbols `isClean` that are not equal.

- Screenshot for `postcss@8.3.0`
![image](https://user-images.githubusercontent.com/34841680/121790658-ca79eb80-cbe1-11eb-8beb-b35878981820.png)

- Screenshot for `postcss@8.3.1`
![image](https://user-images.githubusercontent.com/34841680/121790667-de255200-cbe1-11eb-804d-fb27361f1661.png)

We can use global Symbol via `Symbols.for` in `symbols.js` file in `postcss` this will fix problem with different `postcss` versions, but I don't know is it good decision.
https://github.com/postcss/postcss/blob/2da5501f709862c19e7103b81ba8fb224a5793ff/lib/symbols.js#L3

@ai what do you think about this?

**Comment by ai** [comment]
Wow, great! It looks very obvious now.

I will think about solution. Maybe it is better to switch to string key.

I will mark you as a solver for Cult of Martians task. What is your Twitter account for future `@postcss` tweet?

**Comment by TheSeally** [comment]
Thanks, my twitter link https://twitter.com/autumn_poet

**Comment by ai** [comment]
@TheSeally I made 2 fixes:

1. Long term: require parsers to have `postcss` in `peerDependencies` (but it will take a few weeks to release new versions) https://github.com/postcss/postcss/commit/714c5c6263f5930a5ef0c1704bbc707db9581ed5
2. Short term: detect broken AST and update prototypes https://github.com/postcss/postcss/commit/d8edfeda3804a63d81d010858ade6ce5e49b93a3

Can I ask you to test the second fix?

1. Reproduce error
2. Replace (in standalone `postcss`) `node_modules/postcss/container.js`, `node_modules/postcss/lazy-result.js` and `node_modules/postcss/symbols.js` with a version from GitHub’s `main` branch.
3. Check issue again

**Comment by TheSeally** [comment]
@ai after replacing css is processed correctly

**Comment by ai** [comment]
Fixed with PostCSS 8.3.3.

**Comment by naxxateux** [comment]
Guys, thnx for investigation and fix. Sorry for not responding — I was on vacation :-)
